pulp_rpm/app/downloaders.py

```diff
@@ -0,0 +1,43 @@
+from gettext import gettext as _
+from logging import getLogger
+from aiohttp.client_exceptions import ClientResponseError
+from pulpcore.plugin.download import http_giveup, HttpDownloader
+from urllib.parse import urljoin
+
+log = getLogger(__name__)
+
+
+class SLESHttpDownloader(HttpDownloader):
+    """
+    Custom Downloader that automatically handles authentication token for SLES repositories.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the downloader.
+        """
+        self.auth_token = kwargs.pop('auth_token')
+        super().__init__(*args, **kwargs)
+
+    async def _run(self, extra_data=None):
+        """
+        Download, validate, and compute digests on the `url`. This is a coroutine.
+
+        This method provides the same return object type and documented in
+        :meth:`~pulpcore.plugin.download.BaseDownloader._run`.
+
+        """
+        auth_param = '?' + self.auth_token
+        url = urljoin(self.url, auth_param)
+        async with self.session.get(url) as response:
```

Comment:
It would be better to pass a params dictionary to get() as shown here[0].

[0] https://docs.aiohttp.org/en/stable/client_quickstart.html?highlight=query#passing-parameters-in-urls

Reply:
the token is not in the format of key=value, it is just value so this method is not suitable here.